invenio_pages/resources/resource.py

```diff
@@ -24,36 +25,77 @@ class PageResource(Resource):
 
     def __init__(self, config, service):
         """Constructor."""
-        super(PageResource, self).__init__(config)
+        super().__init__(config)
         self.service = service
 
     def create_url_rules(self):
         """Create the URL rules for the record resource."""
         routes = self.config.routes
         return [
+            route("POST", routes["list"], self.create),
             route("GET", routes["item"], self.read),
+            route("GET", routes["list"], self.search),
+            route("DELETE", routes["item"], self.delete),
+            route("PUT", routes["item"], self.update),
         ]
 
     #
     # Primary Interface
     #
-    @request_extra_args
+    @request_data
+    @response_handler()
+    def create(self):
+        """Create a page."""
+        page = self.service.create(
+            identity=g.identity,
+            data=resource_requestctx.data or {},
+        )
+        if page.to_obj != None:
```

Comment:
with the raise above, you can remove this and simply return:
`return page.to_dict(), 201`